azure/src/tree/AzureAccountTreeItemBase.ts

```diff
@@ -120,19 +119,12 @@ export abstract class AzureAccountTreeItemBase extends AzExtParentTreeItem imple
                         filter.session.tenantId,
                     );
 
-                    // these properties don't exist on TokenCredentials
-                    if (filter.session.credentials2 instanceof DeviceTokenCredentials) {
-                        addExtensionValueToMask(
-                            filter.session.credentials2.clientId,
-                            filter.session.credentials2.domain);
-                    }
-
-                    // these properties don't exist on TokenCredentials
-                    if (filter.session.credentials2 instanceof DeviceTokenCredentials) {
-                        addExtensionValueToMask(
-                            filter.session.credentials2.clientId,
-                            filter.session.credentials2.domain);
-                    }
+                    // these properties don't exist on TokenCredentials, but do exist on DeviceTokenCredentials
+                    // addExtensionValueToMask gracefully handles `undefined` input by ignoring it
+                    addExtensionValueToMask(
```

Comment:
Turns out, we didn't need `instanceof DeviceTokenCredentials` since, if the properties aren't there they will simply be `undefined`, which `addExtensionValueToMask` gracefully handles.

Aside from that, the type was already `TokenCredentialsBase & TokenCredential`, emphasis **&**, which suggests to me it should always be defined anyway. Though if it isn't, that's OK, it will still work (because `addExtensionValueToMask` handles falsy values gracefully).